helm/app/templates/_base_helper.tpl

```diff
@@ -87,6 +87,32 @@ stringData:
 {{- end }}
 {{- end }}
 
+{{- define "service.environment" }}
+{{- with (mergeOverwrite (dict) .service.environment .service.environment_dynamic)}}
+env:
+{{- range $key, $value := . }}
+- name: {{ $key | quote }}
+{{- if kindIs "string" $value }}
+  value: {{ tpl $value $.root | quote }}
+{{- else if kindIs "map" $value }}
+  {{ tpl ($value | toYaml) $.root | nindent 10 }}
```

Comment:
```suggestion
  {{ tpl ($value | toYaml) $.root | nindent 2 }}
```
To account for having nindent applied afterwards

Reply:
thanks, yes sorry

though yaml doesn't care about indenting consistency between different object sub-values, but it'd look weird anyway